### PR TITLE
Deployment spec cleanup

### PR DESCRIPTION
**Pull request by verult**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
* Deployment is now in a standalone namespace rather than `default` to isolate the driver from arbitrary workloads.
* Removed unnecessary driver-registrar role. The events permission is unused
* Bumping snapshotter version to latest
* Changed resizer image to gcr.io
* Added "gce-pd-csi-driver-node" and "gce-pd-csi-driver-controller" priorityClasses and set them in node DaemonSet and controller StatefulSet.
* Changed the label key to "k8s-app" to follow the GKE convention that this deployment is a system app.
* Fixed the name of the resizer RoleBinding

Fixes: #321
Fixes: #37 
Fixes: #357

```release-note
Deployment spec updates:
* The deployment is no longer in namespace `default`
* Changed "app" label key to "k8s-app"
* csi-snapshotter version has been changed to v1.2.0-gke.0
* The resizer role binding has been renamed to "csi-controller-resizer-binding"
* Removed driver-registrar role. 
```

/assign @davidz627 @msau42

### CODE REVIEW

File: deploy/kubernetes/base/setup-cluster.yaml [hunk 1]
```diff
@@ -108,8 +81,32 @@ metadata:
...
   apiGroup: rbac.authorization.k8s.io
+
+---
+
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: csi-controller-leaderelection
+rules:
+- apiGroups: ["coordination.k8s.io"]
```
Comment (davidz627) [inline deploy/kubernetes/base/setup-cluster.yaml]: still necessary?

File: deploy/kubernetes/base/kustomization.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 commonLabels:
-  app: gcp-compute-persistent-disk-csi-driver
+  k8s-app: gcp-compute-persistent-disk-csi-driver
 namespace:
-  default
+  pdcsi-serving
```
Comment (davidz627) [inline deploy/kubernetes/base/kustomization.yaml]: is `gcp-compute-persistent-disk-csi-driver` too wordy? or `pd.csi.storage.gke.io` or some part thereof of either of those?. Basically just any of the names we already have for the driver instead of coming up with with yet another name

**Comment by davidz627** [comment]
Integration tests are failing, driver deployment is failing

**Comment by davidz627** [comment]
Also add: `priorityClassName: system-cluster-critical` and `priorityClassName: system-node-critical` to controller and node YAMLs respectively please

File: deploy/kubernetes/base/setup-cluster.yaml (hunk 1 above)
Comment (verult) [reply deploy/kubernetes/base/setup-cluster.yaml]: Nope, thanks for the catch

File: deploy/kubernetes/base/kustomization.yaml (hunk 2 above)
Comment (verult) [reply deploy/kubernetes/base/kustomization.yaml]: Typing something like `k get po -n pd.csi.storage.gke.io` will be a pain, but worst case I could have an alias so not a huge deal :)

Let me know what you'd prefer the most in the end

Comment (msau42) [reply deploy/kubernetes/base/kustomization.yaml]: gce-pd-csi-driver?

**Comment by verult** [comment]
Addressed comments

File: examples/kubernetes/demo-pod.yaml [hunk 3]
```diff
@@ -1,17 +1,3 @@
-kind: PersistentVolumeClaim
```
Comment (davidz627) [inline examples/kubernetes/demo-pod.yaml]: why delete this?

File: deploy/kubernetes/base/kustomization.yaml [hunk 4]
```diff
@@ -1,7 +1,7 @@
 commonLabels:
-  app: gcp-compute-persistent-disk-csi-driver
+  k8s-app: gcp-compute-persistent-disk-csi-driver
 namespace:
-  default
+  gce-pd-csi-driver
```
Comment (davidz627) [inline deploy/kubernetes/base/kustomization.yaml]: add to readme

File: deploy/kubernetes/base/controller.yaml [hunk 5]
```diff
@@ -14,6 +14,7 @@ spec:
         app: gcp-compute-persistent-disk-csi-driver
     spec:
       serviceAccountName: csi-controller-sa
+      priorityClassName: gce-pd-csi-driver-controller
```
Comment (davidz627) [inline deploy/kubernetes/base/controller.yaml]: add to readme

File: deploy/kubernetes/base/kustomization.yaml [hunk 6]
```diff
@@ -1,7 +1,7 @@
 commonLabels:
-  app: gcp-compute-persistent-disk-csi-driver
+  k8s-app: gcp-compute-persistent-disk-csi-driver
```
Comment (davidz627) [inline deploy/kubernetes/base/kustomization.yaml]: add note to readme

File: examples/kubernetes/demo-pod.yaml (hunk 3 above)
Comment (verult) [reply examples/kubernetes/demo-pod.yaml]: Sorry, an artifact of manual testing stuff. Putting it back

File: deploy/kubernetes/base/kustomization.yaml (hunk 4 above)
Comment (verult) [reply deploy/kubernetes/base/kustomization.yaml]: I don't think the specific namespace name should be documented as it could become out of sync, but I'll mention that the deployment exists in a separate namespace

File: deploy/kubernetes/base/controller.yaml (hunk 5 above)
Comment (verult) [reply deploy/kubernetes/base/controller.yaml]: I don't think the specific namespace name should be documented as it could become out of sync, but I'll mention the behavior of this priorityclass

File: deploy/kubernetes/base/kustomization.yaml (hunk 6 above)
Comment (verult) [reply deploy/kubernetes/base/kustomization.yaml]: I don't think this needs to be documented - it's easy for the entry to be out of date with documentation, and users typically don't have to interact with this label

**Comment by verult** [comment]
Addressed comments.

@davidz627 I think it'd be nice to expand upon the the driver deployment documentation a little more, maybe with an explanation of how the deployment is structured and what to run to deploy/delete the deployment. It should be done outside this PR, but if you think this is useful I can create an issue for it

**Comment by davidz627** [comment]
> It should be done outside this PR, but if you think this is useful I can create an issue for it

Please do. Make sure to capture specific points of difficulty or confusion. Thanks!

**Comment by davidz627** [comment]
/lgtm
/approve
/cc @msau42 
PTAL as well

**Comment by davidz627** [comment]
Looks like this PR broke resize. @verult could you look into that?
